src/9on12Resource.cpp

```diff
@@ -1386,23 +1386,25 @@ namespace D3D9on12
 
         if (RegistryConstants::g_cLockDiscardOptimization)
         {
-            if (mapType == D3D12TranslationLayer::MAP_TYPE_WRITE_DISCARD)
+            if (flags.RangeValid && (mapType == D3D12TranslationLayer::MAP_TYPE_WRITE_DISCARD))

```

Comment:
I think if `flags.RangeValid` isn't set, we need to remove the resource from the map on `WRITE_DISCARD`, don't we? That effectively means that the whole resource is discarded and any future `WRITE_DISCARD`s should be considered overlapping.